test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -493,8 +502,8 @@ class test_uct_cm_sockaddr : public uct_test {
         ASSERT_UCS_OK(status);
         EXPECT_GT(client_port, 0);
 
-        EXPECT_EQ(entity::client_priv_data.length() + 1, remote_data->conn_priv_data_length);
-        EXPECT_EQ(entity::client_priv_data,
+        EXPECT_EQ(entity::client_priv_data.length(), remote_data->conn_priv_data_length);
+        EXPECT_EQ(entity::client_priv_data.c_str(),
```

Comment:
it seems that it can be just `entity::client_priv_data`
if you want to compare `c_str()`, we have to get it from both `std::string` and then use `strcmp()` to compare them